packages/@sanity/cli/src/actions/init-project/bootstrapRemoteTemplate.ts

```diff
@@ -0,0 +1,69 @@
+import {mkdir} from 'node:fs/promises'
+import {join} from 'node:path'
+
+import {debug} from '../../debug'
+import {type CliCommandContext} from '../../types'
+import {
+  applyEnvVariables,
+  applyPackageName,
+  downloadAndExtractRepo,
+  generateSanityApiReadToken,
+  getMonoRepo,
+  isNextJsTemplate,
+  type RepoInfo,
+  validateRemoteTemplate,
+} from '../../util/remoteTemplate'
+import {type GenerateConfigOptions} from './createStudioConfig'
+import {tryGitInit} from './git'
+import {updateInitialTemplateMetadata} from './updateInitialTemplateMetadata'
+
+export interface BootstrapRemoteOptions {
+  outputPath: string
+  repoInfo: RepoInfo
+  bearerToken?: string
+  packageName: string
+  variables: GenerateConfigOptions['variables']
+}
+
+const INITIAL_COMMIT_MESSAGE = 'Initial commit from Sanity CLI'
+
+export async function bootstrapRemoteTemplate(
```

Comment:
Clearly seeing the pipeline of things that need to be done serially here but unsure how failure of this half-way would look. We have a pattern of showing the error messages in the cli but if this partially completes, it might be confusing to users to know how to proceed if they have the repo down but were unable to have any of the files edited. Perhaps some try/catch around groups of this where the errors are still thrown to be printed on the cli but have some surrounding context like "Unable to edit newly downloaded files in [folder]/. See documentation on updating files from a template: some-docs link" or something else could be helpful in getting people un-stuck at certain points.

Reply:
If the template does not pass validation, we will just abort, and you should get a message with why it did not pass. This is fine imo.

There are a new other steps it could fail in after we've downloaded files over to the users computer.

1. Applying environment variables: For this I've gone ahead and added a more descriptive error message with a link to our environment variable docs article.
2. Applying package name: I've renamed this function to tryApplyPackageName, and will noop on catch block, as this is not a big deal if fails.
3. Git init also runs tryGitInit and will noop if it fails.